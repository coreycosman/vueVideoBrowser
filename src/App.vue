<template>
  <div data-test="app">
    <SearchBar @termChange="onTermChange"> </SearchBar>
    <div class="two-column">
      <VideoList
        @videoObjectToApp="setVideoObjectProp"
        :videos="videos"
      ></VideoList>
      <VideoDetail :selectedVideo="selectedVideo"></VideoDetail>
    </div>
  </div>
</template>

<script>
import axios from "axios";
import SearchBar from "./components/SearchBar";
import VideoList from "./components/VideoList";
import VideoDetail from "./components/VideoDetail";

const VIDEO_KEY = process.env.VUE_APP_VIDEO_KEY;

export default {
  name: "App",
  components: { SearchBar, VideoList, VideoDetail },
  data() {
    return {
      videos: [],
      selectedVideo: {}
    };
  },
  methods: {
    async onTermChange(searchTerm) {
      try {
        const res = await axios.get(
          "https://www.googleapis.com/youtube/v3/search",
          {
            params: {
              key: VIDEO_KEY,
              type: "video",
              part: "snippet",
              q: searchTerm
            }
          }
        );
        this.videos = res.data.items;
      } catch (error) {
        console.log(error);
      }
    },

    setVideoObjectProp(video) {
      this.selectedVideo = video;
    }
  }
};
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}

.two-coulmn {
  display: flex;
}
</style>
