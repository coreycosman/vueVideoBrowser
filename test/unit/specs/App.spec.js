import { shallowMount, mount } from "@vue/test-utils";
import App from "@/App";
import axios from "axios";
jest.mock("axios");

describe("App.vue", () => {
  beforeEach(() => {
    let wrapper = mount(App);
  });
  const wrapper = mount(App);
  it("should mount App with data props", () => {
    expect(wrapper.vm.videos).toMatchSnapshot();
    expect(wrapper.vm.selectedVideo).toMatchSnapshot();
  });

  it("should set videos state to axios array response when onTermChange is called", async () => {
    let wrapper = shallowMount(App);
    const res = {
      data: {
        items: ["video1", "video2", "video3", "video4", "video5"]
      }
    };
    axios.get.mockResolvedValue(res);
    await wrapper.vm.onTermChange("video search term");

    expect(wrapper.vm.videos).toEqual(res.data.items);
  });
  it("should render html with all child components", () => {
    expect(wrapper.find("[data-test='app']").text()).toMatchSnapshot();
    expect(wrapper.vm.$children[0].$options._componentTag).toEqual("SearchBar");
    expect(wrapper.vm.$children[1].$options._componentTag).toEqual("VideoList");
    expect(wrapper.vm.$children[2].$options._componentTag).toEqual(
      "VideoDetail"
    );
  });
});
