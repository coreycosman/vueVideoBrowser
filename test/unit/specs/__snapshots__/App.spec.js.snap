// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should mount App with data props 1`] = `Array []`;

exports[`App.vue should mount App with data props 2`] = `Object {}`;

exports[`App.vue should render html with all child components 1`] = `""`;
